packages/help-center/src/hooks/use-article-content-click.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, type MouseEvent } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const isThisASupportArticleLink = ( href: string ) =>
+	/wordpress\.com(\/\w\w)?(?=\/support\/)|support\.wordpress\.com/.test( href );
+
+export const useArticleContentClick = ( node: HTMLDivElement | null ) => {
+	const navigate = useNavigate();
+
+	const filters = [
+		/**
+		 * Make support article links open within Help Center.
+		 */
+		{
+			condition: ( target: HTMLElement ) =>
+				target.tagName === 'A' && isThisASupportArticleLink( target.getAttribute( 'href' ) || '' ),
+			action: ( event: MouseEvent< HTMLElement > ) => {
+				event.preventDefault();
+				const target = event.target as HTMLElement;
+				const href = target.getAttribute( 'href' );
+				navigate( `/post?link=${ href }` );
+			},
+		},
+
+		/**
+		 * Fix jump-to-section links.
+		 */
+		{
+			condition: ( target: HTMLElement ) =>
+				target.tagName === 'A' && target.getAttribute( 'href' )?.startsWith( '#' ),
```

Comment:
Maybe `matches` would be cleaner?
```suggestion
				target.matches( 'a[href^=#]' )
```